### PR TITLE
SAMBRO: CAP History Model + SAMBRO: fixes

### PR DESCRIPTION
**Pull request by biplovbhandari**
@flavour Please merge

### CODE REVIEW

File: controllers/cap.py [hunk 1]
```diff
@@ -1391,4 +1403,106 @@ def cap_AreaRowOptionsBuilder(alert_id, caller=None):
 
         return cap_area_options
 
+# -----------------------------------------------------------------------------
+def compose():
+
+    if settings.has_module("msg") and \
+       settings.get_cap_show_notification_menu() and \
+       auth.permission.has_permission("update", c="cap", f="alert"):
+        if request.get_vars.option == "notify_subscribers":
```
Comment (flavour) [inline controllers/cap.py]: Can just use get_vars in the controller context :)

File: controllers/cap.py [hunk 2]
```diff
@@ -1391,4 +1403,106 @@ def cap_AreaRowOptionsBuilder(alert_id, caller=None):
 
         return cap_area_options
 
+# -----------------------------------------------------------------------------
+def compose():
```
Comment (flavour) [inline controllers/cap.py]: Did you try using msg.compose() ?

File: views/cap/compose.html [hunk 3]
```diff
@@ -0,0 +1,2 @@
+{{extend "layout.html"}}
+{{include "cap/_compose.html"}}
```
Comment (flavour) [inline views/cap/compose.html]: I don't think you gain value from separating these files within a single theme
- the point of the separation is to allow multiple themes

Comment (flavour) [inline views/cap/compose.html]: I'm not sure why CAP needs a separate Compose function & View?
What is different about the CAP one?

Sure there are some permission checks etc as a wrapper to the compose form, but the form itself should be standard & hence best DRY?

**Comment by biplovbhandari** [comment]
@flavour corrected! Please merge

**Pull request merged**